lib/ansible/modules/cloud/dimensiondata/dimensiondata_network.py

```diff
@@ -159,181 +162,239 @@
 
 
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.dimensiondata import get_credentials, DimensionDataAPIException, LibcloudNotFound
+from ansible.module_utils.dimensiondata import DimensionDataModule, DimensionDataAPIException
 from ansible.module_utils.pycompat24 import get_exception
 try:
     from libcloud.compute.types import Provider
     from libcloud.compute.providers import get_driver
     from libcloud.compute.base import NodeLocation
+
     import libcloud.security
+
     HAS_LIBCLOUD = True
 except ImportError:
     HAS_LIBCLOUD = False
 
 
-def network_obj_to_dict(network, version):
-    network_dict = dict(id=network.id, name=network.name,
-                        description=network.description)
-    if isinstance(network.location, NodeLocation):
-        network_dict['location'] = network.location.id
-    else:
-        network_dict['location'] = network.location
-
-    if version == '1.0':
-        network_dict['private_net'] = network.private_net
-        network_dict['multicast'] = network.multicast
-        network_dict['status'] = None
-    else:
-        network_dict['private_net'] = None
-        network_dict['multicast'] = None
-        network_dict['status'] = network.status
-    return network_dict
-
-
-def get_mcp_version(driver, location):
-    # Get location to determine if MCP 1.0 or 2.0
-    location = driver.ex_get_location_by_id(location)
-    if 'MCP 2.0' in location.name:
-        return '2.0'
-    return '1.0'
-
-
-def create_network(module, driver, mcp_version, location,
-                   name, description):
-
-    # Make sure service_plan argument is defined
-    if mcp_version == '2.0' and 'service_plan' not in module.params:
-        module.fail_json('service_plan required when creating network and ' +
-                         'location is MCP 2.0')
-    service_plan = module.params['service_plan']
-
-    # Create network
-    try:
-        if mcp_version == '1.0':
-            res = driver.ex_create_network(location, name,
-                                           description=description)
-        else:
-            res = driver.ex_create_network_domain(location, name,
-                                                  service_plan,
-                                                  description=description)
-    except DimensionDataAPIException:
-        e = get_exception()
-
-        module.fail_json(msg="Failed to create new network: %s" % str(e))
+class DimensionDataNetworkModule(DimensionDataModule):
+    """
+    The dimensiondata_network module for Ansible.
+    """
+
+    def __init__(self):
+        """
+        Create a new Dimension Data network module.
+        """
+
+        super(DimensionDataNetworkModule, self).__init__(
+            module=AnsibleModule(
+                argument_spec=dict(
+                    region=dict(
```

Comment:
I started off like that, but it looked difficult to read (to me, at least). I think it's the use of `dict` instead of `{}`. But it that's the preferred style I'm happy to change it :)